src/release-pr.ts

```diff
@@ -340,4 +406,111 @@ export class ReleasePR {
   addPath(file: string) {
     return ReleasePR.addPathStatic(file, this.path);
   }
+
+  // BEGIN release functionality
+
+  // Override this method to detect the release version from code (if it cannot be
+  // inferred from the release PR head branch)
+  protected async detectReleaseVersionFromCode(): Promise<string | undefined> {
+    return undefined;
```

Comment:
I don't think we're ready to expose this yet, or even if it's a good idea at all

Reply:
This would be the hook for looking up the version(s) from the manifest.

Reply:
ah, I interpreted `FromCode` to mean per-type specific source code (e.g. version.py, package.json)

I'd prefer to wait on this change till we have something actually ready to implement it. I'd actually like to propose that we use a eventually only use a manifest file for tracking versions. I'd add it as part of the aggregate PR work, providing a migration path that seamlessly switches from encoding the version in the branch name to writing it to the manifest.

For now I feel like we should stick with the current functionality of version-in-branch-name